File: app/src/main/java/dev/danielholmberg/improve/Components/Tag.java
package dev.danielholmberg.improve.Components;

import java.io.Serializable;

public class Tag implements Serializable {

    private String tagId;
    private String label;
    private String colorHex;
    private int colorInt;

    public Tag() {}

    public Tag(String tagId, String label, String colorHex, int colorInt) {
        this.tagId = tagId;
        this.label = label;
        this.colorHex = colorHex;
        this.colorInt = colorInt;
    }

    public String getTagId() {
        return tagId;
    }

    public void setTagId(String tagId) {
        this.tagId = tagId;
    }

    public String getLabel() {
        return label;
    }

    public void setLabel(String label) {
        this.label = label;
    }

    public String getColorHex() {
        return colorHex;
    }

    public void setColorHex(String color) {
        this.colorHex = color;
    }

    public int getColorInt() {
        return colorInt;
    }

    public void setColorInt(int colorInt) {
        this.colorInt = colorInt;
    }
}

File: app/src/main/java/dev/danielholmberg/improve/Models/Tag.kt
package dev.danielholmberg.improve.Models;

import android.os.Parcel;
import android.os.Parcelable;

public class Tag implements Parcelable {

    private String id;
    private String label;
    private String color;
    private String textColor;

    public Tag() {}

    public Tag(String id) {
        this.id = id;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getLabel() {
        return label;
    }

    public void setLabel(String label) {
        this.label = label.toUpperCase();
    }

    public String getColor(){
        return this.color;
    }

    public void setColor(String color) {
        this.color = color;
    }

    public String getTextColor() {
        return this.textColor;
    }

    public void setTextColor(String textColor) {
        this.textColor = textColor;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(id);
        parcel.writeString(label);
        parcel.writeString(color);
    }

    protected Tag(Parcel in) {
        this.id = in.readString();
        this.label = in.readString();
        this.color = in.readString();
    }

    public static final Creator<Tag> CREATOR = new Creator<Tag>() {
        @Override
        public Tag createFromParcel(Parcel in) {
            return new Tag(in);
        }

        @Override
        public Tag[] newArray(int size) {
            return new Tag[size];
        }
    };
}
